AntistasiOfficial.Altis/Functions/fn_changeCitySupply.sqf

```diff
@@ -0,0 +1,70 @@
+
+
+
+params["_type", "_delta", "_location"];
+private["_city", "_data", "_supplyLevels", "_index", "_currentLevel", "_newLevels"];
+
+//waitUntil{!citySupplyIsChanging}
+//citySupplyIsChanging = true;
+
+//Gather the needed data
+_city = [[ciudades, _location] call BIS_fnc_nearestPosition, _location] select (typeName _location == typeName "");
+_data = server getVariable _city;
+if !(_data isEqualType []) exitWith {diag_log format ["Error in changeCitySupply. Passed %1 as reference.", _location]};
+
+//Execute command
+_supplyLevels = _data select 4;
+
+if(_type == "FOOD") then {_index = 0;};
+if(_type == "WATER") then {_index = 1;};
+if(_type == "FUEL") then {_index = 2;};
+
+_currentLevel = _supplyLevels select _index;
+
+if(_delta > 0 && _currentLevel != 'GOOD') then
+{
+	if(_currentLevel == 'LOW') then
+	{
+		_currentLevel = 'GOOD';
+		[0,5,_city] remoteExec ["AS_fnc_changeCitySupport",2];
+	};
+	
+	if(_currentLevel == 'CRITICAL') then
+	{
+		_currentLevel = 'LOW';
+		[0,15,_city] remoteExec ["AS_fnc_changeCitySupport",2];
+	};
+};
+if(_delta < 0 && _currentLevel != 'CRITICAL') then
+{
+	if(_currentLevel == 'GOOD') then
+	{
+		_currentLevel = 'LOW';
+	};
+	if(_currentLevel == 'LOW') then
+	{
+		_currentLevel = 'CRITICAL';
+	};
+};
+
+//Create new Array containing the new data
+_newLevels = [];
+for "_i" from 0 to 2 do
+{
+	if(_i != _index) then
```

Comment:
_index  is undefined here... probably a scope issue, i see private above.. uhmm not sure how to fix

Reply:
_index is defined line 18 to 20
```
if(_type == "FOOD") then {_index = 0;};
if(_type == "WATER") then {_index = 1;};
if(_type == "FUEL") then {_index = 2;};```

if undefined it means that _type is any of this this values;
You can either add default _index value above these conditions
or
fix the call of this function by passing a proper value to _type